components/script/dom/bluetooth.rs

```diff
@@ -100,6 +79,46 @@ impl Clone for AllowedBluetoothDevice {
     }
 }
 
+#[derive(HeapSizeOf, JSTraceable)]
+pub struct BluetoothExtraPermissionData {
+    permission_data: DOMRefCell<BluetoothPermissionData>,
```

Comment:
We should inline the members of BluetoothPermissionData inside BluetoothExtreaPermissionData; there's no need for the extra level of indirection.